service/controller/clusterapi/v28/key/machine_deployment.go

```diff
@@ -1,29 +1,72 @@
 package key
 
 import (
+	"math/bits"
+	"net"
 	"strconv"
 
 	g8sv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/ipam"
 	"github.com/giantswarm/microerror"
 	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 
+	"github.com/giantswarm/aws-operator/pkg/annotation"
 	"github.com/giantswarm/aws-operator/pkg/label"
 )
 
 func WorkerClusterID(cr v1alpha1.MachineDeployment) string {
 	return cr.Labels[label.Cluster]
 }
 
+func WorkerSubnet(cr v1alpha1.MachineDeployment) string {
+	return cr.Annotations[annotation.MachineDeploymentSubnet]
+}
+
 // TODO this method has to be properly implemented and renamed eventually.
-func StatusAvailabilityZones(cr v1alpha1.MachineDeployment) []g8sv1alpha1.AWSConfigStatusAWSAvailabilityZone {
+func StatusAvailabilityZones(cr v1alpha1.MachineDeployment) ([]g8sv1alpha1.AWSConfigStatusAWSAvailabilityZone, error) {
+	var workerSubnet net.IPNet
+	{
+		s := WorkerSubnet(cr)
+		if s == "" {
+			return nil, microerror.Maskf(notFoundError, "MachineDeployment is missing subnet allocation")
+		}
+
+		_, n, err := net.ParseCIDR(s)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		workerSubnet = *n
+	}
+
 	var azs []g8sv1alpha1.AWSConfigStatusAWSAvailabilityZone
-	for _, s := range WorkerAvailabilityZones(cr) {
-		azs = append(azs, g8sv1alpha1.AWSConfigStatusAWSAvailabilityZone{
-			Name: s,
-		})
+	{
+		azsSubnets, err := splitNetwork(workerSubnet, uint(len(WorkerAvailabilityZones(cr))))
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		for i, s := range WorkerAvailabilityZones(cr) {
+			subnets, err := splitNetwork(azsSubnets[i], 2)
```

Comment:
Split cluster's per-AZ subnet into public and private subnet for present in-TCCP-workers structure. This will change once the workers are extracted out to separate CF stack.